modules/page_base.py

```diff
@@ -420,40 +440,46 @@ def fill(
         el.send_keys(f"{term}{end}")
         return self
 
-    def multi_click(
-        self, iters: int, reference: Union[str, tuple, WebElement], labels=[]
-    ) -> Page:
-        """Perform multiple clicks at once on an element by name, selector, or WebElement"""
-        if self.context == "chrome":
-            self.set_chrome_context()
+    def fetch(self, reference: Union[str, tuple, WebElement], labels=[]) -> WebElement:
+        """
+        Given an element name, a selector, or a WebElement, return the
+        corresponding WebElement.
+        """
         if isinstance(reference, str):
-            el = self.get_element(reference, labels=labels)
+            return self.get_element(reference, labels=labels)
         elif isinstance(reference, tuple):
-            el = self.find_element(**reference)
+            return self.find_element(*reference)
         elif isinstance(reference, WebElement):
-            el = reference
-        else:
-            assert False, "Attempted to multiclick on something unsupported"
+            return reference
+        assert False, "Attempted to multiclick on something unsupported"
 
-        # Little cheat: if element doesn't exist in one context, try the other
-        n = 0
-        while n < 2:
-            try:
-                n += 1
-                if iters == 2:
-                    self.actions.double_click(el).perform()
```

Comment:
I'm a bit confused at this block of code, why is it one of them uses `double_click` but the other just clicks it?

Reply:
Also, instead of having a while loop with 2 iterations, why not just do a try catch? so the try tries to multi click in the desired context and then the catch tries the other? Just wondering